src/components/DateRangeInput/DateRangeInput.tsx

```diff
@@ -197,14 +209,25 @@ const DateRangeInput: React.FC<
     [setCurrentRange, currentDateRange]
   );
 
-  const isDisabled = useCallback(
-    () =>
-      !currentDateRange[0] ||
-      !currentDateRange[1] ||
-      currentDateRange[0].isAfter(currentDateRange[1], withTime ? 'minute' : 'day'),
-    [currentDateRange]
-  );
-
+  const onClear = useCallback(() => setCurrentRange([undefined, undefined]), [setCurrentRange]);
+
+  const isDisabled = React.useMemo(() => {
+    if (areDatesUndefined(previousDateRange) && areDatesUndefined(currentDateRange)) {
+      return true;
+    }
+    if (currentDateRange[0] && !currentDateRange[1]) {
+      return true;
+    }
+
+    return currentDateRange.every((date, index) => {
+      return (
+        previousDateRange &&
+        previousDateRange[index] &&
```

Comment:
I  think you can fully  omit the first 2 checks. The value of `previousDateRange` is the output of  `datesToDayjs` which is always an array. Doing `date?.isSame(undefined)` will yield `false` so we are ok, no?

Reply:
Not always, essentially we have some business logic baked in.
As users are browsing and selecting a date range they might come into a situation where they might pick just the first date from the date range, in this case, an error will be thrown. I the same manner  `previousDateRange` might be undefined throwing an error as well.